File: backend/src/main/java/com/tl/models/application/game/ws_messages/messages/UserUpdateMessage.java
package com.tl.models.application.game.ws_messages.messages;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.tl.models.application.game.ws_messages.Message;
import com.tl.models.application.game.ws_messages.message_type.MessageType;
import com.tl.models.application.user.User;
import lombok.*;

import java.util.List;

public class UserUpdateMessage extends Message {
    @JsonProperty
    private List<User> users;

    public UserUpdateMessage(List<User> users) {
        super(MessageType.UserUpdate);
        this.users = users;
    }

    @Override
    @SneakyThrows
    public String serialize() {
        return new ObjectMapper().writeValueAsString(this);
    }
}

File: frontend/dog-web/src/app/models/websockets/user-update-message.ts
import { User } from "../http/user";
import { MessageTypeMapper } from "./dto/message-type.dto";
import { UserUpdateMessageDTO } from "./dto/user-update-message.dto";
import { Message } from "./message";
import { MessageType } from "./message-type";

export class UserUpdateMessage extends Message {
    constructor(
        type: MessageType, 
        public users: User[]
    ) {
        super(type);
    }

    public static fromApi({type, users}: UserUpdateMessageDTO): UserUpdateMessage {
        return new UserUpdateMessage(MessageTypeMapper.fromApi(type), users.map(User.fromApi))
    }

}